internal/cli/template_create.go

```diff
@@ -0,0 +1,183 @@
+package cli
+
+import (
+	"os"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+type ProjectTemplateCreateCommand struct {
+	*baseCommand
+
+	flagSummary                    string
+	flagExpandedSummary            string
+	flagReadmeMarkdownTemplatePath string
+	flagWaypointHCLTemplatePath    string
+	flagTFCNoCodeModuleSource      string
+	flagTFCNoCodeModuleVersion     string
+	flagTags                       []string
+}
+
+func (c *ProjectTemplateCreateCommand) Run(args []string) int {
+	flagSet := c.Flags()
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(flagSet),
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+	args = flagSet.Args()
+	ctx := c.Ctx
+
+	if len(args) != 1 {
+		c.ui.Output("Single argument required.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	name := args[0]
+
+	var template pb.ProjectTemplate
+
+	template.Name = name
+	if c.flagSummary != "" {
+		template.Summary = c.flagSummary
+	}
+	if c.flagExpandedSummary != "" {
+		template.ExpandedSummary = c.flagExpandedSummary
+	}
+
+	if c.flagReadmeMarkdownTemplatePath != "" {
+		rmt, err := os.ReadFile(c.flagReadmeMarkdownTemplatePath)
+		if err == os.ErrNotExist {
+			rmt = []byte("")
+		} else if err != nil {
+			c.ui.Output("Unable to read readme.md template file: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+		template.ReadmeMarkdownTemplate = rmt
+	}
+
+	if c.flagWaypointHCLTemplatePath != "" {
+		wpt, err := os.ReadFile(c.flagWaypointHCLTemplatePath)
+		if err != nil {
+			c.ui.Output("Unable to read waypoint.hcl template file: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+		template.WaypointProject = &pb.ProjectTemplate_WaypointProject{
+			WaypointHclTemplate: wpt,
+		}
+	}
+
+	if c.flagTFCNoCodeModuleSource != "" && c.flagTFCNoCodeModuleVersion == "" {
```

Comment:
Should we also have an `if` case here where both the module source and the No Code module version are empty?

Also, it might be worth allowing the user to save an incomplete project template, where not all of the necessary fields are added. We can chalk that up to a future consideration though, not necessarily for this PR!

Reply:
These might be two conflicting options. Currently a user can either specify both tfc module and version or nothing at all and both of those options would be valid. I'd argue that if a user is putting in a terraform module source, they should already know the version and vice versa.